### ISSUE TITLE
Fix `property-no-unknown` false positive for `font-display`

### ISSUE THREAD

**Issue by Mister-Hope** [stylelint/stylelint#9115]
### What minimal example or steps are needed to reproduce the bug?

```
mister-hope@mac-mini-studio ecosystem % pnpm lint

> @vuepress/ecosystem@ lint /Users/mister-hope/projects/ecosystem
> oxlint --fix --type-aware --type-check --report-unused-disable-directives --deny-warnings && oxfmt && stylelint **/*.{css,html,scss,vue} --fix

Found 0 warnings and 0 errors.
Finished in 2.3s on 1144 files with 401 rules using 10 threads.

tools/helper/src/client/styles/normalize.scss
  17:3  ✖  Unexpected unknown property "font-display"  property-no-unknown

✖ 1 problem (1 error, 0 warnings)

 ELIFECYCLE  Command failed with exit code 2.
mister-hope@mac-mini-studio ecosystem % pnpm stylelint -v
17.4.0
```

<img width="895" height="144" alt="Image" src="https://github.com/user-attachments/assets/8da00158-a0ac-492f-a623-4a8f24a50726" />

I think this bug might be related to https://github.com/stylelint/stylelint/pull/9099

Note: everything is totally fine in 17.2.0

**Comment by romainmenke** [comment]
Hi @Mister-Hope,

Can you provide sample code?
The code in your screenshot is partially obscured.

----

I quickly checked with the [Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACAzCA7ALgWhQQzBgAJgAddIotLbKASwGcAHAGzwE94iInM6M8LANzkAvuXIALTAFsWJcpWo56zNp269+6QSPSiQAGhAo6LGADk8MuIhgAPa6xgA6MAwZHwGUwHMEIA5MEABOmESw+ACuLOFkFEQOmDDoUAxcANoA5AyY7OYsdDSQ6H7YuXipeCFQWQC6hmJ6XiV+AGKhMniYAbn5MIVYbj50vi4yAFaexrBMnojxlKQgfQVFmMtcy2zJucuNCcurA+vYraPlmJVQ1VCbRNvdMHsgYuQgokA) and found that `font-display` only gives an error when you use it in a style rule, a rule with a selector.

This is expected as `font-display` is only valid as a descriptor in `@font-face`.

```css
@font-face {
  font-display: optional; /* no error, as expected */
}

html {
  font-display: optional; /* errors, but also expected */
}
```

Is it possible that Stylelint is accurately pointing out an issue in your source code?

**Comment by jeddy3** [comment]
Thanks @romainmenke, I was just writing the same thing (like yesterday 😆 ).

@Mister-Hope [Specification for reference](https://drafts.csswg.org/css-fonts/#font-display-desc).

It was a false negative bug that we fixed in [17.4.0](https://stylelint.io/changelog/#1740---2026-02-25).

**Comment by Mister-Hope** [comment]
Thanks for the clarification:)
